helpers/training/trainer.py

```diff
@@ -2228,7 +2280,26 @@ def train(self):
                             f"Received {bsz} latents, but expected {self.config.train_batch_size}. Processing short batch."
                         )
                     training_logger.debug(f"Working on batch size: {bsz}")
-                    if self.config.flow_matching:
+                    if self.config.model_family == "omnigen":
+                        # x1 corresponds to your latents
+                        x1 = latents
+
+                        # Sample x0 from a standard normal distribution with the same shape as latents
+                        x0 = torch.randn_like(latents)
+
+                        # Sample t for each sample in the batch using the specified distribution
+                        u = torch.randn(bsz, device=latents.device)
+                        t = 1 / (1 + torch.exp(-u))  # t ∈ (0, 1)
+                        t = t.to(latents.device, dtype=latents.dtype)
+
+                        # Convert t to timesteps compatible with the model (scaled appropriately)
+                        timesteps = t * 999
+                        timesteps = timesteps.to(
+                            self.accelerator.device, dtype=latents.dtype
+                        )
+                        timesteps = timesteps.long()
```

Comment:
@staoxiao also mostly copied your noise sampling in at this point

Reply:
@bghira, we use a float value between 0-1 for timesteps, so there is no need to scale it `timesteps = t * 999`